sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/declarative/DeclarativeEndpointInfoTransformer.java

```diff
@@ -103,7 +115,34 @@ public static ConnectorInfo createAndAddConnectorInfo(
   }
 
   private static String readDocumentation(String defaultDocsPath, String path, String id) {
-    final var resourcePath = path.isEmpty() ? String.format("%s/%s.md", defaultDocsPath, id) : path;
+    if (path.isEmpty()) {
+      return findFileByIdAndGetContent(defaultDocsPath, id);
+    }
+    return getSpecifiedFileContent(path);
+  }
+
+  private static String findFileByIdAndGetContent(String defaultDocsPath, String id) {
+    try {
+      List<Path> resources = getResourcesFromClasspath(defaultDocsPath);
+
+      Optional<Path> file =
+          resources.stream()
+              .filter(
+                  resource ->
+                      resource.getFileName().toString().equalsIgnoreCase(id + MARKDOWN_EXTENSION))
+              .findFirst();
+
+      if (file.isPresent()) {
+        Path filePath = file.get();
+        return Files.readString(filePath);
+      }
+      return null;
+    } catch (IOException | URISyntaxException e) {
+      throw SIPFrameworkException.init("Failed to read documentation resource", e);
```

Comment:
Maybe a bit more declarative exception message. You have id of declarative strucuture element, you can use it in the message.